AdminServer/appscale/admin/__init__.py

```diff
@@ -254,37 +254,17 @@ def version_from_payload(self):
 
     return version
 
-  def project_exists(self, project_id):
-    """ Checks if a project exists.
-    
-    Args:
-      project_id: A string specifying a project ID.
-    Raises:
-      CustomHTTPError if unable to determine if project exists.
-    """
-    try:
-      return self.ua_client.does_app_exist(project_id)
-    except UAException:
-      message = 'Unable to check if project exists: {}'.format(project_id)
-      logging.exception(message)
-      raise CustomHTTPError(HTTPCodes.INTERNAL_ERROR, message=message)
+  def version_exists(self, project_id, service_id, version_id):
+    """ Checks if a version exists.
 
-  def create_project(self, project_id, runtime):
-    """ Creates a new project.
-    
     Args:
       project_id: A string specifying a project ID.
-      runtime: A string specifying the project's runtime.
-    Raises:
-      CustomHTTPError if unable to create new project.
+      service_id: A string specifying a service ID.
+      version_id: A string specifying a version ID.
     """
-    logging.info('Creating project: {}'.format(project_id))
-    try:
-      self.ua_client.commit_new_app(project_id, runtime)
-    except UAException:
-      message = 'Unable to ensure project exists: {}'.format(project_id)
-      logging.exception(message)
-      raise CustomHTTPError(HTTPCodes.INTERNAL_ERROR, message=message)
+    version_node = '/appscale/projects/{}/services/{}/versions/{}'.format(
+      project_id, service_id, version_id)
+    return self.zk_client.exists(version_node) is not None
```

Comment:
I'm not familiar with the client: can it thrown an exception if it cannot talk to zookeeper?

Reply:
Yes, it can throw a ZooKeeper exception. We could catch it so that we can return a properly-formatted JSON error. I think it should be a separate pull request, because we don't catch that exception for any of our ZooKeeper calls in this package, yet.